koin-projects/koin-android-viewmodel/src/main/java/org/koin/android/viewmodel/ext/android/ActivityExt.kt

```diff
@@ -29,41 +30,38 @@ import kotlin.reflect.KClass
  *
  * @author Arnaud Giuliani
  */
-
-/**
- * LifecycleOwner extensions to help for ViewModel
- *
- * @author Arnaud Giuliani
- */
-
-fun <T : ViewModel> ViewModelStoreOwner.viewModel(
-    clazz: KClass<T>,
+inline fun <reified T : ViewModel> AppCompatActivity.viewModel(
     qualifier: Qualifier? = null,
-    parameters: ParametersDefinition? = null
+    noinline owner: ViewModelOwnerDefinition = { from(this) },
+    noinline parameters: ParametersDefinition? = null
 ): Lazy<T> {
-    return lazy(LazyThreadSafetyMode.NONE) { getViewModel(clazz, qualifier, parameters) }
+    return lazy(LazyThreadSafetyMode.NONE) {
+        getViewModel<T>(qualifier, owner, parameters)
+    }
 }
 
-inline fun <reified T : ViewModel> ViewModelStoreOwner.viewModel(
+fun <T : ViewModel> AppCompatActivity.viewModel(
```

Comment:
Why do you need to replace `ViewModelStoreOwner` with `AppCompatActivity`? Maybe it will be better to keep original function and add separate one that will be extension for `ComponentActivity`?

Reply:
yes, both can help 👍

Reply:
`ComponentActivity` is only available in AndroidX